### ISSUE TITLE
Welcome to test OpenDeepFaceSwap

### ISSUE THREAD

**Issue by iperov** [deepfakes/faceswap-playground#134]
https://github.com/iperov/OpenDeepFaceSwap

**Comment by feepdakenoob** [comment]
Trying it now.

**Comment by iperov** [comment]
@feepdakenoob are you on linux ?

first you need to compile DLIB 19.10 with cuda

**Comment by feepdakenoob** [comment]
windows

**Comment by feepdakenoob** [comment]
Do I need to start entire workflow from scratch or can I use existing alignments etc?

**Comment by iperov** [comment]
yes start from scratch.

**Comment by iperov** [comment]
for windows I will release prebuilt binaries with zero dependencies at all, except video drivers

**Comment by seranus** [comment]
can you write a quick summary of your models pros/cons or comparison?

**Comment by HelpSeeker** [comment]
Quick question: Isn't keras-contrib also a dependency (at least if you want to use the MTCNN detector)? If yes, could you please mention it somewhere?

**Comment by iperov** [comment]
@HelpSeeker keras_contrib used in df and iae

**Comment by iperov** [comment]
@seranus ah quick summary...

sorted by best ascending and vram usage:

H64 (2GB+) 
F128 (3GB+) 
DF (4GB+)
IAEF128 (4GB+)
SIAEF128  (6GB+)

**Comment by seranus** [comment]
@iperov thanks, so h64 is lowmem and origianl, df is fakers, what model is f128?

**Comment by iperov** [comment]
same as h64 but full face 128

**Comment by iperov** [comment]
h64 not only low mem. Its automatically adjust NN dims is vram gb < 4

**Comment by iperov** [comment]
changed loss for mask decoder to 'mae' , restart training.

**Comment by iperov** [comment]
@dfaker I found that 'mae' loss for mask making crisp edge with less artifacts, while 'mse' making smooth edge.

**Comment by ruah1984** [comment]
@iperov , all this short form stand for what. i will try one them this morning 
H64 (2GB+)
F128 (3GB+)
DF (4GB+)- Dfaker
IAEF128 (4GB+)
SIAEF128 (6GB+)

**Comment by iperov** [comment]
added H128 - half face 128.

**Comment by gessyoo** [comment]
Testing latest commit, extract --detector mt, on Win 10 x64, Anaconda 3.6, TF 1.7, GTX 1070, dlib and opencv compiled with CUDA 9.0.  About 88% CPU load and 10-15% GPU load, running first extract pass.  I have one GPU, but it reports running extract on four GTX 1070 devices.

**Comment by iperov** [comment]
its fine

**Comment by fengzhong22** [comment]
iperov,对于Windows，除了视频驱动程序之外，我将发布预编译的二进制文件，其零依赖性。现在有发布吗？

**Comment by iperov** [comment]
@fengzhong22 soon

**Comment by Sonofdemoc** [comment]
@iperov Is there any way to use an existing face set?
Good dst faceset is really hard to make.
People will not leave the original and will not create a new face set.

any idea?

**Comment by feepdakenoob** [comment]
@Sonofdemoc As models and encoders evolve you are going to have to throw away and restart. That's how it goes. Keep your source vids/frames around.

**Comment by iperov** [comment]
> Now I'm just extracting the extracted faceset
dont do this.

**Comment by feepdakenoob** [comment]
Dos the space bar do anything in the preview window? 

And do you have any examples of the different models? Like, what's an ideal loss to aim for or what batch sizes to use? 

So far I'm playing with LIAEF128 and it's very interesting stuff

**Comment by iperov** [comment]
@feepdakenoob space bar currently not affects preview, because all models generates only one preview page.

I am working on examples to repo. Ideal loss ? less - better. Batch size is automatic. Decrease it manually if train didnt start.

**Comment by feepdakenoob** [comment]
@iperov Excellent results after training over night. Probably the best so far.

I really, really like how the alignments are in the png. That makes life so much easier.

**Comment by tjess78** [comment]
@iperov Great optimization of your LIAEF128 model!! LIAEF128 works now with higher batch size than H128 on a GTX 1060 6gb.

last build:
H128: 16
LIAEF128: 2
 
actual build:
H128: 16
LIAEF128: 32

**Comment by sonofdemoc2** [comment]
> It sounds silly. Why should I deceive you? for what? if it's not a face-swap, where on earth did those results come from? And I am not alone reporting the problem. Are you going to ban everyone who says there's a problem?
> 
> I don't know what a fool this is.
> 
> You tell me to go away, so I'm off.
> 
> But what you're doing is a big mistake. Everyone watched this.
> 
> -------------------------------------------------------------------------------------------------

https://github.com/iperov/OpenDeepFaceSwap/issues/32

@iperov Even censorship? you made it up. I can put my workspace up again and when people test it, they know you're lying. You can't erase this. people will know.

**Comment by iperov** [comment]
I dont want to spend more time to proof something. You fooled me. You trained faceswap on different faceset than odfs.

**Comment by sonofdemoc2** [comment]
@iperov I am training back to 20,000 epoch again from the beginning. After that, I will upload my workspace together. People will soon find out who is cheating. 

You're deleting and closing then making me liar? I don't know what you're up to, but I can't let you get away.

**Comment by iperov** [comment]
I trained your faceset on faceswap and result eyes same as odfs.
My goal only make odfs better. But you provided false comparison, which cause huge lost of my time.
I dont care what you will post to ppl in foreign repos. Absolutely.
